### PR TITLE
[ENH] move experimental surface plotting into the stable API

### PR DESCRIPTION
**Pull request by Remi-Gau**
- Closes none but relates to #4568 

Changes proposed in this pull request:

- adapt the new surface plotting functions to use the new SurfaceImage
- update doc strings
- move tests for surface plotting input checks
- leave "shim" in the experimental module to be removed after the examples have been moved

### CODE REVIEW

File: nilearn/plotting/html_surface.py [hunk 1]
```diff
@@ -433,9 +435,10 @@ def view_img_on_surf(
 
 @fill_doc
 def view_surf(
-    surf_mesh,
+    surf_mesh=None,
```
Comment (Remi-Gau) [inline nilearn/plotting/html_surface.py:438]: Warning: this is technically a change in the API.

File: nilearn/plotting/surf_plotting.py [hunk 2]
```diff
@@ -729,7 +731,7 @@ def _plot_surf_matplotlib(
 
 @fill_doc
 def plot_surf(
-    surf_mesh,
+    surf_mesh=None,
```
Comment (Remi-Gau) [inline nilearn/plotting/surf_plotting.py:734]: Warning: this is technically a change in the API.

File: nilearn/plotting/surf_plotting.py [hunk 3]
```diff
@@ -1040,8 +1064,9 @@ def _get_faces_on_edge(faces, parc_idx):
 
 @fill_doc
 def plot_surf_contours(
-    surf_mesh,
-    roi_map,
+    surf_mesh=None,
+    roi_map=None,
+    hemi=None,
```
Comment (Remi-Gau) [inline nilearn/plotting/surf_plotting.py:1069]: Warning: this is technically a change in the API.

File: nilearn/plotting/surf_plotting.py [hunk 4]
```diff
@@ -1217,8 +1283,8 @@ def plot_surf_contours(
 
 @fill_doc
 def plot_surf_stat_map(
-    surf_mesh,
-    stat_map,
+    surf_mesh=None,
+    stat_map=None,
```
Comment (Remi-Gau) [inline nilearn/plotting/surf_plotting.py:1287]: Warning: this is technically a change in the API.

File: nilearn/experimental/plotting/_surface_plotting.py
Comment (Remi-Gau) [inline nilearn/experimental/plotting/_surface_plotting.py:1]: these functions are now just 'redirects' to the stable API and are just there while we wait for all the examples to be moved.

File: nilearn/plotting/html_surface.py (hunk 1 above)
Comment (bthirion) [reply nilearn/plotting/html_surface.py:438]: Yes, but I don't think it is harmful. Maybe mention it explicitly in the changelog.

**Review by bthirion — COMMENTED** [review]
Besides the changelog, I see nothing to add to this PR.

File: doc/Makefile [hunk 5]
```diff
@@ -108,6 +108,12 @@ html-noplot: sym_links_datasets glm_reports
 	@echo
 	@echo "Build finished. The HTML pages are in $(BUILDDIR)/html."
 
+html-noplot-noreport: sym_links_datasets
```
Comment (Remi-Gau) [inline doc/Makefile:111]: recipe to build doc faster locally by not generating GLM and maskers reports

**Review by man-shu — APPROVED** [review]
LGTM!

**Comment by Remi-Gau** [comment]
Thanks @bthirion  and @man-shu for the cross-checks

**Pull request merged**